components/layout/display_list/mod.rs

```diff
@@ -785,11 +795,13 @@ impl Fragment {
                 rect.origin.y += font_metrics.ascent - font_metrics.underline_offset;
                 rect.size.height =
                     Au::from_f32_px(font_metrics.underline_size.to_nearest_pixel(dppx));
+
                 self.build_display_list_for_text_decoration(
                     &parent_style,
                     builder,
                     &rect,
                     text_decoration,
+                    TextDecorationSecondDoubleLienBias::Down,
```

Comment:
These callers could also pass the relevant `TextDecorationLine` (`TextDecorationLine::UNDERLINE` here) to avoid the need for the new `TextDecorationSecondDoubleLineBias` struct, but fine either way.